HLTrigger/Muon/plugins/HLTMuonL3orL2PreFilter.cc

```diff
@@ -0,0 +1,343 @@
+      min_Nhits_L2_(iConfig.getParameter<std::vector<int> >("MinNhitsL2")),
+      min_NmuonHits_L2_(iConfig.getParameter<int>("MinNmuonHitsL2")),
+      absetaBins_L2_(iConfig.getParameter<std::vector<double> >("AbsEtaBinsL2")),
+      minNstations_L2_(iConfig.getParameter<std::vector<int> >("MinNstationsL2")),
+      minNchambers_L2_(iConfig.getParameter<std::vector<int> >("MinNchambersL2")),
+      matchPreviousCand_(iConfig.getParameter<bool>("MatchToPreviousCand")) {
+  candToken_ = consumes<reco::RecoChargedCandidateCollection>(candTag_);
+  L2CandToken_ = consumes<reco::RecoChargedCandidateCollection>(L2CandTag_);
+  previousCandToken_ = consumes<trigger::TriggerFilterObjectWithRefs>(previousCandTag_);
+  beamspotToken_ = consumes<reco::BeamSpot>(beamspotTag_);
+}
+
+HLTMuonL3orL2PreFilter::~HLTMuonL3orL2PreFilter() = default;
+
+//
+// member functions
+//
+void HLTMuonL3orL2PreFilter::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  makeHLTFilterDescription(desc);
+  desc.add<edm::InputTag>("BeamSpotTag", edm::InputTag("hltOfflineBeamSpot"));
+  desc.add<edm::InputTag>("CandTag", edm::InputTag("hltL3MuonCandidates"));
+  desc.add<edm::InputTag>("L2CandTag", edm::InputTag("hltL2MuonCandidates"));
+  desc.add<edm::InputTag>("PreviousCandTag", edm::InputTag(""));
+  desc.add<int>("MinN", 1);
+  desc.add<double>("MaxDz", 9999.0);
+  desc.add<double>("MinDxySig", -1.0);
+  desc.add<double>("NSigmaPt", 0.0);
+  desc.add<double>("MaxDXYBeamSpot", 9999.0);
+  desc.add<double>("MinDXYBeamSpot", -1.0);
+  desc.add<double>("MaxPtDifference", 9999.0);
+  desc.add<double>("MinTrackPt", 0.0);
+  desc.add<bool>("MatchToPreviousCand", true);
+  desc.add<double>("MaxEtaL3", 2.5);
+  desc.add<double>("MinPtL3", 3.0);
+  desc.add<double>("MaxNormalizedChi2L3", 9999.0);
+  desc.add<int>("MinNmuonHitsL3", 0);
+  desc.add<int>("MinNhitsL3", 0);
+  desc.add<double>("MaxEtaL2", 2.5);
+  desc.add<double>("MinPtL2", 3.0);
+  desc.add<double>("MaxNormalizedChi2L2", 9999.0);
+  desc.add<int>("MinNmuonHitsL2", 0);
+  desc.add<std::vector<double> >("AbsEtaBinsL2", std::vector<double>(1, 9999.));
+  desc.add<std::vector<int> >("MinNstationsL2", std::vector<int>(1, 1));
+  desc.add<std::vector<int> >("MinNhitsL2", std::vector<int>(1, 0));
+  desc.add<std::vector<int> >("MinNchambersL2", std::vector<int>(1, 0));
+  descriptions.add("hltMuonL3orL2PreFilter", desc);
+}
+
+// ------------ method called to produce the data  ------------
+bool HLTMuonL3orL2PreFilter::hltFilter(edm::Event& iEvent,
+                                       const edm::EventSetup& iSetup,
+                                       trigger::TriggerFilterObjectWithRefs& filterproduct) const {
+  // All HLT filters must create and fill an HLT filter object,
+  // recording any reconstructed physics objects satisfying (or not)
+  // this HLT filter, and place it in the Event.
+
+  if (saveTags())
+    filterproduct.addCollectionTag(candTag_);
```

Comment:
I think you need to add also ```filterproduct.addCollectionTag(L2CandTag_);```
as you also want to record L2cands.

Reply:
Yes sorry, I will add it